ply-core/src/Ply/Core/Types.hs

```diff
@@ -38,6 +39,8 @@ import UntypedPlutusCore (DeBruijn, DefaultFun, DefaultUni, Program)
 import Ply.Core.Serialize.Script (serializeScriptCbor)
 import Ply.Core.Typename (Typename)
 
+data AsData a = AsData a
```

Comment:
Is there a particular reason why this is not a `newtype`?

Reply:
Nice catch. I was planning on different thing.

Reply:
this should be caught by the linter, this should be part of pre-commit-hooks. I will probably open a PR with some nix fixes if I find time...